tests/decomposition/test_decomposition_graph.py

```diff
@@ -87,15 +138,16 @@ def test_graph_construction(self, _):
         """Tests constructing a graph from a single Hadamard."""
 
         op = qml.Hadamard(wires=[0])
-        graph = DecompositionGraph(operations=[op], gate_set={"RX", "RZ", "GlobalPhase"})
+
+        graph = DecompositionGraph(operations=[op], gate_set={"RX": 1.0, "RZ": 1.0, "GlobalPhase": 1.0})
         # 5 ops and 3 decompositions (2 for Hadamard and 1 for RY) and 1 dummy starting node
         assert len(graph._graph.nodes()) == 9
         # 8 edges from ops to decompositions, 3 from decompositions to ops, and 3 from the
         # dummy starting node to the target gate set.
         assert len(graph._graph.edges()) == 14
 
         # Check that graph construction stops at gates in the target gate set.
-        graph2 = DecompositionGraph(operations=[op], gate_set={"RY", "RZ", "GlobalPhase"})
+        graph2 = DecompositionGraph(operations=[op], gate_set={"RY": 1.0, "RZ": 1.0, "GlobalPhase": 1.0})
```

Comment:
Same comment here. The weights only affect how the graph is solved, not how the graph is constructed.